setup.py

```diff
@@ -203,6 +204,51 @@ def run(self):
             sys.exit(128)
 
 
+class Test(SimpleCommand):
+    """Run selftests"""
+
+    description = 'Run selftests'
+    user_options = [
+        ("job-api", None, "Run job API checks"),
+        ("static-checks", None, "Run static checks (isort, lint, etc)"),
+        ("nrunner-interface", None, "Run selftests/functional/test_nrunner_interface.py"),
+        ("unit", None, "Run selftests/unit/"),
+        ("jobs", None, "Run selftests/jobs/"),
+        ("functional", None, "Run selftests/functional/"),
+        ("optional-plugins", None, "Run optional_plugins/*/tests/"),
+        ("disable-plugin-checks", None, "Disable checks for a plugin (by directory name)"),
+        ("features", None, "Show the features tested by this test")
```

Comment:
nitpick: maybe s/features/list-features/g or s/features/show-features/g

Reply:
> nitpick: maybe s/features/list-features/g or s/features/show-features/g

Nope, this is also following `check.py`.

Reply:
IMO, we are creating a targets options to setup.py, I don't think we have to restricted to what is inside check.py.

Reply:
Ok, I was revisiting this thread, take this as example.

I understand that here we passing the same arguments to `check.py`. But we also agree that `check.py` is not a good example, so IMO we don't have to replicate this model to the new `setup.py` targets, otherwise, when changing `check.py` latter we will need to change this too, creating a bad user experience.

My suggestion:

```python
mapping = {'list-features': 'features'}
...
args.features = mapping.get(self.features)
selftests.check.main(args)
```

So, in summary, let's think about good names without being influenced by `check,py` so we are free to change `check.py` latter without affecting users/developers.

Reply:
Given that the issue is not only about *listing* the features, but about selecting them, we'd need at the very least:

```setup.py test --list-features```

AND

```setup.py test --features=unit,functional,...```

Which is more flexible indeed, and allows for more maneuvering around `check.py` current state. But, it's also worse usability wise (users have to run `--list-features` first, and then with the valid options).

Having said that, I don't think it's worth implementing this logic of listing/enabling "features" into `setup.py test` itself. I believe its place is really in an Avocado job (`avocado.core.job.Job`) and with `features` meaning the `suites` of a job.

Reply:
Even better... +1 for not having 'list/show' actions here.

Reply:
FIY, this is the issue about the ideas I listed above: https://github.com/avocado-framework/avocado/issues/4832